libtcmu-register.c

```diff
@@ -190,21 +185,27 @@ tcmulib_reg_name_vanished(GDBusConnection *connection,
 			  const gchar     *name,
 			  gpointer         user_data)
 {
-	tcmu_err("Failed to get bus %s\n", name);
+	struct tcmulib_context *ctx = user_data;
+
+	ctx->connection = NULL;
 }
 
 void tcmulib_register(struct tcmulib_context *ctx)
 {
-	assert(!ctx->reg_count_down);
-	ctx->reg_count_down = darray_size(ctx->handlers);
-	if (!ctx->reg_count_down)
-		return;
+	struct tcmulib_handler *handler;
+	
+	/* Start acquiring buses for each subtype owned by this context. */
+	darray_foreach(handler, ctx->handlers) {
+		tcmulib_handler_own_bus(handler);
+	}
+
+	/* Start waiting for the primary service to become available */
 	g_bus_watch_name(G_BUS_TYPE_SYSTEM,
-			 "org.kernel.TCMUService1",
-			 G_BUS_NAME_WATCHER_FLAGS_NONE,
-			 tcmulib_reg_name_appeared,
-			 tcmulib_reg_name_vanished,
-			 ctx,
-			 NULL);
+		"org.kernel.TCMUService1",
+		G_BUS_NAME_WATCHER_FLAGS_NONE,
+		tcmulib_reg_name_appeared,
+		tcmulib_reg_name_vanished,
+		ctx,
+		NULL);
 
```

Comment:
Sorry for the delay on this one. Just noticed my comment never went through (mobile interface I was using on vacation is different from normal browser and put my comment in pending).

I am fine with this patch for now. My only comment was to just revert this coding style chance as the existing is more standard for the tcmu-runner code.

Reply:
This was unintentional, fixed this and a couple of white space problems.